packages/concerto-core/lib/introspect/stringvalidator.js

```diff
@@ -37,17 +38,45 @@ class StringValidator extends Validator{
      * Create a StringValidator.
      * @param {Object} field - the field or scalar declaration this validator is attached to
      * @param {Object} validator - The validation string. This must be a regex
+     * @param {Object} lengthValidator - The length validation string - [minLength,maxLength] (inclusive).
      *
      * @throws {IllegalModelException}
      */
-    constructor(field, validator) {
+    constructor(field, validator, lengthValidator) {
         super(field, validator);
-        try {
-            const CustomRegExp = field?.parent?.getModelFile()?.getModelManager()?.options?.regExp || RegExp;
-            this.regex = new CustomRegExp(validator.pattern, validator.flags);
+        this.minLength = null;
+        this.maxLength = null;
+        this.regex = null;
+
+        if (lengthValidator) {
+            if(Object.prototype.hasOwnProperty.call(lengthValidator, 'minLength')) {
+                this.minLength = lengthValidator.minLength;
+            }
+            if(Object.prototype.hasOwnProperty.call(lengthValidator, 'maxLength')) {
+                this.maxLength = lengthValidator.maxLength;
+            }
+            if(this.minLength === null && this.maxLength === null) {
+                // can't specify no upper and lower value
+                this.reportError(null, 'Invalid string length, minLength and-or maxLength must be specified.');
+            } else if (this.minLength < 0 || this.maxLength < 0) {
+                this.reportError(null, 'minLength and-or maxLength must be positive integers.');
+            } else if (this.minLength === null || this.maxLength === null) {
+                // this is fine and means that we don't need to check whether minLength > maxLength
```

Comment:
Is this necessary if there is no code executed on the condition evaluating to true?

Reply:
I copied most of the logic from number validator. I overlooked it may be. Let me remove it